From: "Rafael J. Wysocki" <rafael.j.wysocki@intel.com>
Date: Tue, 25 Nov 2025 15:05:01 +0100
Subject: Revert "ACPI: processor: Remove unused empty stubs of some functions"
Patch-mainline: v6.18
Git-commit: 1a8b3501821b608383f7c7aa0f24e2006681e2b5
References: jsc#PED-14260

Revert commit 5020d05b3476 ("ACPI: processor: Remove unused empty stubs
of some functions") because it depends on a problematic one.

Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
Acked-by: Chun-Yi Lee <jlee@suse.com>
---
 include/acpi/processor.h |   20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

--- a/include/acpi/processor.h
+++ b/include/acpi/processor.h
@@ -425,6 +425,26 @@ int acpi_processor_power_state_has_chang
 int acpi_processor_hotplug(struct acpi_processor *pr);
 void acpi_processor_register_idle_driver(void);
 void acpi_processor_unregister_idle_driver(void);
+#else
+static inline int acpi_processor_power_init(struct acpi_processor *pr)
+{
+	return -ENODEV;
+}
+
+static inline int acpi_processor_power_exit(struct acpi_processor *pr)
+{
+	return -ENODEV;
+}
+
+static inline int acpi_processor_power_state_has_changed(struct acpi_processor *pr)
+{
+	return -ENODEV;
+}
+
+static inline int acpi_processor_hotplug(struct acpi_processor *pr)
+{
+	return -ENODEV;
+}
 #endif /* CONFIG_ACPI_PROCESSOR_IDLE */
 
 /* in processor_thermal.c */
